### PR TITLE
Add new search operators

### PR DESCRIPTION
**Pull request by rajkambo**
Just creating a draft PR to keep track of progress.

Other operators for other searchTypes coming soon 👍

### CODE REVIEW

**Comment by ManjotS** [comment]
please use  camel case with  no _ for var names..

**Comment by rajkambo** [comment]
> please use camel case with no _ for var names..

👍 even for constants?

**Comment by ManjotS** [comment]
I think so.. whatever is the js norm

File: api/lib/searchOperators.js [hunk 1]
```diff
@@ -1,4 +1,171 @@
...
+        condition: "json_extract(v.Translations, '$.en.bdb') LIKE ?",
+        parameters: [modifiedSearchQuery],
+      };
+    }
+    return {
+      columns: ' LEFT JOIN tokenized_english t ON t.verseid = v.ID',
+      condition: 't.token LIKE ?',
+      // shouldn't there be a % at the beginning as well?
+      // lets just keep it the same for now to not break existing flows.
+      parameters: [`${searchQuery}%`],
```
Comment (rajkambo) [inline api/lib/searchOperators.js]: I wasn't quite sure why they only appended the '%' to the end..

File: api/lib/searchOperators.js [hunk 2]
```diff
@@ -1,4 +1,171 @@
...
+        condition: 'v.Gurmukhi LIKE BINARY ?',
+        parameters: [modifiedSearchQuery],
+      };
+    }
+    return {
+      condition: 'v.Gurmukhi LIKE BINARY ?',
+      parameters: [`%${modifiedSearchQuery}%`],
+    };
+  },
+  fullWordEnglishToQuery: searchQuery => {
```
Comment (rajkambo) [inline api/lib/searchOperators.js]: tbh this function is very similar to the above `fullWordGurmukhiToQuery`. I think we can definitely abstract them out to cut down on the amount of code as a future improvement.

File: api/lib/searchOperators.js (hunk 1 above)
Comment (rajkambo) [reply api/lib/searchOperators.js]: oh nvm its because its tokenized.

**Comment by rajkambo** [comment]
Alright so this pretty much covers it for searchType=0, searchType=2,searchType=3, and searchType=6.

Full word is good for testing with operators, and so is "first letter start".

Overall, there is a bunch of improvements that could be used from a code reuse perspective, but we should have a decent enough implementation of the mentioned searchTypes to do some testing 👍

File: api/lib/searchOperators.js (hunk 2 above)
Comment (ManjotS) [reply api/lib/searchOperators.js]: agree

**Pull request merged**